api/v1alpha1/statuscheck_types.go

```diff
@@ -0,0 +1,215 @@
+	StatusCheckContinuous StatusCheckMode = "Continuous"
+)
+
+type StatusCheckType string
+
+const (
+	TypeHTTP StatusCheckType = "HTTP"
+)
+
+type StatusCheckSpec struct {
+	// Mode defines the execution mode of the status check.
+	// Support type: Synchronous / Continuous
+	// +optional
+	// +kubebuilder:validation:Enum=Synchronous;Continuous
+	Mode StatusCheckMode `json:"mode,omitempty"`
+
+	// Type defines the specific status check type.
+	// Support type: HTTP
+	// +kubebuilder:default=HTTP
+	// +kubebuilder:validation:Enum=HTTP
+	Type StatusCheckType `json:"type"`
+
+	// Duration defines the duration of the status check.
+	// A duration string is a possibly signed sequence of
+	// decimal numbers, each with optional fraction and a unit suffix,
+	// such as "300ms", "-1.5h" or "2h45m".
+	// Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
+	// +optional
+	Duration *string `json:"duration,omitempty" webhook:"Duration"`
+
+	// TimeoutSeconds defines the number of seconds after which the status check times out.
+	// +optional
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Minimum=1
+	TimeoutSeconds int `json:"timeoutSeconds,omitempty"`
+
+	// PeriodSeconds defines how often (in seconds) to perform the status check.
+	// +optional
+	// +kubebuilder:default=10
+	// +kubebuilder:validation:Minimum=1
+	PeriodSeconds int `json:"periodSeconds,omitempty"`
+
+	// FailureThreshold defines when a status check fails,
+	// it will try FailureThreshold times before giving up.
+	// +optional
+	// +kubebuilder:default=3
+	// +kubebuilder:validation:Minimum=1
+	FailureThreshold int `json:"failureThreshold,omitempty"`
+
+	// RecordsHistoryLimit defines the number of record to retain.
+	// +optional
+	// +kubebuilder:default=100
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=1000
+	RecordsHistoryLimit int `json:"recordsHistoryLimit,omitempty"`
+
+	// +optional
+	*EmbedStatusCheck `json:",inline,omitempty"`
+}
+
+type StatusCheckStatus struct {
+	// StartTime represents time when the status check started to execute.
+	// +optional
+	StartTime *metav1.Time `json:"startTime,omitempty"`
+
+	// CompletionTime represents time when the status check was completed.
+	// +optional
+	CompletionTime *metav1.Time `json:"completionTime,omitempty"`
+
+	// Count represents the total number of the status check executed.
+	// +optional
+	Count int64 `json:"count,omitempty"`
+
+	// Conditions represents the latest available observations of a StatusCheck's current state.
+	// +optional
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	Conditions []StatusCheckCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+
+	// Records contains the history of the execution of StatusCheck.
+	// +optional
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	Records []StatusCheckRecord `json:"records,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
```

Comment:
I think we could remove `patchMergeKey` and `patchStrategy`, because we do not need the feature about merging when patch this object

Reply:
oops, yes, I forgot to remove these line.

Reply:
Updated.